docs/architecture-decisions/2024-03-06_local_dev_ui_packaging.md

```diff
@@ -0,0 +1,62 @@
+# Local dev UI packaging
+
+- **Status**: Draft
+- **Owner:** Patrick Dinh (MakerX)
+- **Deciders**: Alessandro (Algorand Foundation), Rob Moore (MakerX), MakerX team
+- **Date created**: 2024-03-06
+
+## Context
+
+We are building a web-based local explorer interface to support:
+
+- Transaction / asset / account explorer
+- Dev wallet integration (including KMD support)
+- Network switching
+- ARC-32 app studio
+- Visual transaction representations and basic visual debugging interface to help new users understand what’s happening
+- AlgoKit init and Algokit tasks CLI exploration user interfaces to help new users get the most out of the CLI
+
+The explorer will be packaged as a executable as well as deployed as a website.
+
+## Requirements
+
+- The explorer should support a wide variety of Linux distributions, macOS (both Apple Silicon and Intel architectures), and Windows.
+- The explorer can be deployed to a website. The explorer website has limited functionalities, compared to the desktop version.
+
+## Options
+
+### Option 1 - Election
+
+### Option 2 - Tauri
+
+[Tauri](https://tauri.app/about/intro) is a toolkit that helps developers make applications for the major desktop platforms - using virtually any frontend framework in existence. The core is built with Rust, and the CLI leverages Node.js making Tauri a genuinely polyglot approach to creating and maintaining great apps.
+
+**Pros**
+Tauri supports all intended operations for the local dev UI MVP via their JavaScript API without the need to write any Rust.
+
+- It can manage the [file systems](https://tauri.app/v1/api/js/fs), launch another [process](https://tauri.app/v1/api/js/process), run a [shell command](https://tauri.app/v1/api/js/shell).
+- Tauri integrates well with major web frameworks. [`create-tauri-app`](https://github.com/tauri-apps/create-tauri-app) can bootstrap a working app within minutes.
+  - Once bootstrapped, the web app can be bundled individually and deployed as a website. Below is the `npm script` Tauri generates for a Vite project, we can see that it supports `vite build`
```

Comment:
Neat that it supports Vite since it's such a good local dev experience and pretty standard web approach these days so there should be broad skillset support for it in industry.